roscala/src/main/scala/coop/rchain/rosette/prim/tuple.scala

```diff
@@ -88,6 +88,36 @@ object tuple {
     }
   }
 
+  /**
+    * Define the "tuple-safe-nth" primitive.
+    * This safely returns the nth element of the specified Tuple
+    * e.g. (tuple-safe-nth [1 2 3 4 5] 3) ==> 4
+    */
+  object tplSafeNth extends Prim {
+    override val name: String = "tuple-safe-nth"
+    override val minArgs: Int = 2
+    override val maxArgs: Int = 2
+
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Fixnum] = {
+      val elem = ctxt.argvec.elem
+
+      checkTuple(0, elem).right.flatMap( // Ensure arg0 is a Tuple
```

Comment:
`Either` is right-biased so you should not need `right` here.

Reply:
Done